src/src/DataStructs/SettingsStruct.h

```diff
@@ -175,6 +175,14 @@ class SettingsStruct_tmpl
   // Try to extend settings to make the checksum 4-byte aligned.
 //  uint8_t       ProgmemMd5[16]; // crc of the binary that last saved the struct to file.
 //  uint8_t       md5[16];
+#ifdef HAS_ETHERNET
+  uint8_t       ETH_Phy_Addr;
+  int8_t        ETH_Pin_mdc;
+  int8_t        ETH_Pin_mdio;
+  int8_t        ETH_Pin_power;
+  int8_t        ETH_Phy_Type;
+  uint8_t       ETH_Clock_Mode;
```

Comment:
Make sure these values do get some proper initialization values.
This means for existing settings you need to initialize them to proper defaults (e.g. 0)
and increase the internal BUILD value (is now 20105) and also init it to proper values when initializing new settings.
Would be nice if you can fit these values in a 32 bit value, so we don't get padding of the settings.
A 32-bit value is always 4-byte aligned, but 8-bit values are not.
You can then make a get/set function in the struct to handle these parameters.

Reply:
Hmm i can do some bit-manipulations to use only one uint32_t, if i understood you correctly.

I will also add ESP32-PoE to DeviceModel, so it could be possible to set pre-defined configuration.

But may i ask you to give a hint where are defined initial values for SettingsStruct in the source? Generic, not DeviceModel based. I was looking for them

Reply:
I'm not really sure why in this case it's better to put this values in 32bit?
As i can see, there are a lot of uint8_t or 16t or even booleans in SettingsStruct right now

Reply:
For existing members you must not change their signature as it may also affect their position in the struct.
For new members you will add at the end, you determine where the next member will be positioned.

A "large" member will be positioned in memory at an address which is a multiple of its own size (max modulo 4 bytes on this platform)
For example, a struct with these members:
```
char
uint32_t
char
uint32_t
```
will be 16 bytes, while it can be 10 bytes if you order them by size, largest first.

Reply:
you don't need to change anything here, @TD-er just explained how align work on esp

Reply:
Okay, i know about struct alignment in C, but i just couldn't find reason how this could be inefficient in case of 6 8-bit values ;)

Reply:
You don't need 6x 8 bits, so you could store it in 32 bit :)

But maybe it is a bit too much effort for at most 2 bytes gain in settings struct size.
Just leave it.